### PR TITLE
Include content-length header for S3 GET responses

### PR DESCRIPTION
**Pull request by Arun-LinkedIn**
S3 Java client seem to be expecting content-length header for GET requests. This change is to include it for S3 requests

### CODE REVIEW

File: ambry-frontend/src/main/java/com/github/ambry/frontend/AmbrySecurityService.java [hunk 1]
```diff
@@ -394,7 +395,8 @@ private void setGetBlobResponseHeaders(BlobInfo blobInfo, GetBlobOptions options
       restResponseChannel.setHeader(RestUtils.Headers.CONTENT_RANGE, rangeAndLength.getFirst());
       contentLength = rangeAndLength.getSecond();
     }
-    if (contentLength < frontendConfig.chunkedGetResponseThresholdInBytes) {
+    if (contentLength < frontendConfig.chunkedGetResponseThresholdInBytes || RestUtils.isS3Request(restRequest)) {
```
Comment (justinlin-linkedin) [inline ambry-frontend/src/main/java/com/github/ambry/frontend/AmbrySecurityService.java:398]: I would prefer to add content length header in the S3GetBlobHandler so we don't add any special cases here.

Comment (Arun-LinkedIn) [reply ambry-frontend/src/main/java/com/github/ambry/frontend/AmbrySecurityService.java]: Sure. Updated to have the logic in S3GetBlobHandler

**Review by SophieGuo410 — APPROVED** [review]
lgtm!

File: ambry-frontend/src/main/java/com/github/ambry/frontend/s3/S3GetHandler.java [hunk 2]
```diff
@@ -80,7 +85,18 @@ protected void doHandle(RestRequest restRequest, RestResponseChannel restRespons
     } else if (S3BaseHandler.isMultipartListPartRequest(restRequest)) {
       s3MultipartUploadHandler.handle(restRequest, restResponseChannel, callback);
     } else {
-      getBlobHandler.handle(requestPath, restRequest, restResponseChannel, callback);
+      getBlobHandler.handle(requestPath, restRequest, restResponseChannel,
+          buildCallback(frontendMetrics.s3GetHandleMetrics, (r) -> {
```
Comment (justinlin-linkedin) [inline ambry-frontend/src/main/java/com/github/ambry/frontend/s3/S3GetHandler.java:89]: you don't need (r), just 
```
r -> {
    xxxx
}
```

**Review by justinlin-linkedin — APPROVED** [review]

**Pull request merged**